packages/react-utilities/src/hooks/useEvent.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+/**
+ * Hook used to add and remove an event from an element.
+ *
+ * @param element - The element to listen to.
+ * @param eventName - The event to watch.
+ * @param callback - The callback to call when the eventName is triggered.
+ * @param useCapture - Whether the events should be dispatched to the callback first before other EventTargets.
+ */
+export const useEvent = <T>(
```

Comment:
With the different ways to add event handlers to DOM elements in React, I'm not sure if we should be looking for a silver bullet solution. Here are a few different examples how this can happen:

1. 'normal' case, the element always exists in DOM and the listeners should be applied for the duration of the component lifecycle
```tsx
const ref = React.useRef(null);

React.useEffect(() => {
  if (ref.current) {
    el.addEventListener
  }

  return () => el.removeEventListener
}, [ref])
```
2. Add/cleanup event listener on certain dependencies/flags
```tsx
const ref = React.useRef(null);

React.useEffect(() => {
  if (ref.current && otherCondition) {
    el.addEventListener
  }

  return () => el.removeEventListener
}, [ref, otherCondition])
```

3. callback ref if we want to make sure the event listener changes if the underlying DOM node changes, this can't be done with `useRef` that easily because ref changes don't trigger rerenders
```tsx
const callbackRef = React.useCallback(el => {
  if (el) {
    el.addEventListener
  } else {
    el.removeEventListener
  }
})
```

Reply:
I totally agree with @ling1726 at this one, seems to me like this is an overkill that doesn't grasp all possible scenarios and I don't see a lot of benefits to justify this.